incubator/spark/templates/spark-worker-deployment.yaml

```diff
@@ -34,3 +34,8 @@ spec:
             requests:
               cpu: "{{.Values.Worker.Cpu}}"
               memory: "{{.Values.Worker.Memory}}"
+          env:
+          - name: SPARK_DAEMON_MEMORY
+            value: {{ default "1g" .Values.Worker.DaemonMemory | quote }}
+          - name: SPARK_WORKER_MEMORY
+            value: {{ default "1g" .Values.Worker.WorkerMemory | quote }}
```

Comment:
Great catch! Thanks @prydonius. Will fix. I didn't catch it in the lint because the default was getting rendered